webclient/grpc/src/main/java/io/helidon/webclient/grpc/GrpcClientCall.java

```diff
@@ -0,0 +1,191 @@
+
+import io.grpc.CallOptions;
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+
+/**
+ *  * An implementation of a gRPC call. Expects:
+ * <p>
+ * start (request | sendMessage)* (halfClose | cancel)
+ *
+ * @param <ReqT> request type
+ * @param <ResT> response type
+ */
+class GrpcClientCall<ReqT, ResT> extends GrpcBaseClientCall<ReqT, ResT> {
+    private static final Logger LOGGER = Logger.getLogger(GrpcClientCall.class.getName());
+
+    private final ExecutorService executor;
+    private final AtomicInteger messageRequest = new AtomicInteger();
+
+    private final LinkedBlockingQueue<BufferData> sendingQueue = new LinkedBlockingQueue<>();
+    private final LinkedBlockingQueue<BufferData> receivingQueue = new LinkedBlockingQueue<>();
+
+    private final CountDownLatch startReadBarrier = new CountDownLatch(1);
+    private final CountDownLatch startWriteBarrier = new CountDownLatch(1);
+
+    private volatile Future<?> readStreamFuture;
+    private volatile Future<?> writeStreamFuture;
+
+    GrpcClientCall(GrpcClientImpl grpcClient, MethodDescriptor<ReqT, ResT> methodDescriptor, CallOptions callOptions) {
+        super(grpcClient, methodDescriptor, callOptions);
+        this.executor = grpcClient.webClient().executor();
+    }
+
+    @Override
+    public void request(int numMessages) {
+        LOGGER.finest(() -> "request called " + numMessages);
+        messageRequest.addAndGet(numMessages);
+        startReadBarrier.countDown();
+    }
+
+    @Override
+    public void cancel(String message, Throwable cause) {
+        LOGGER.finest(() -> "cancel called " + message);
+        responseListener().onClose(Status.CANCELLED, EMPTY_METADATA);
+        readStreamFuture.cancel(true);
+        writeStreamFuture.cancel(true);
+        close();
+    }
+
+    @Override
+    public void halfClose() {
+        LOGGER.finest("halfClose called");
+        sendingQueue.add(EMPTY_BUFFER_DATA);       // end marker
+    }
+
+    @Override
+    public void sendMessage(ReqT message) {
+        LOGGER.finest("sendMessage called");
+        BufferData messageData = BufferData.growing(BUFFER_SIZE_BYTES);
+        messageData.readFrom(requestMarshaller().stream(message));
+        BufferData headerData = BufferData.create(5);
+        headerData.writeInt8(0);                                // no compression
+        headerData.writeUnsignedInt32(messageData.available());         // length prefixed
+        sendingQueue.add(BufferData.create(headerData, messageData));
+        startWriteBarrier.countDown();
+    }
+
+    protected void startStreamingThreads() {
+        // write streaming thread
+        writeStreamFuture = executor.submit(() -> {
+            try {
+                startWriteBarrier.await();
+                LOGGER.fine("[Writing thread] started");
+
+                boolean endOfStream = false;
+                while (isRemoteOpen()) {
+                    LOGGER.finest("[Writing thread] polling sending queue");
```

Comment:
For sockets, I would like to always include the socket id in the output. There is a method on socket context to log a message that will do that for you in a way consistent with other log statements (so we can correlate log output from multiple areas)

Reply:
Will check that